test.l

```diff
@@ -1,31 +1,34 @@
 #!/usr/bin/env bin/lumen
 
+(when-compiling
+  (define-macro test (x msg)
+    `(if (not ,x)
+         (do (set failed (+ failed 1))
+             (return ,msg))
+       (inc passed)))
+
+  (define-macro test= (a b)
+    (let-unique (x y)
+      `(let (,x ,a ,y ,b)
+         (test (equal? ,x ,y)
+               (cat "failed: expected " (str ,x) ", was " (str ,y))))))
+
+  (define-macro define-test (name rest: body)
+    `(add tests (list ',name (fn () ,@body))))
+
+  nil)
```

Comment:
We can also modify all of the `define` macros to return their name, quoted. That would eliminate the need for the trailing `nil` in the `when-compiling` statement, since

```
(when-compiling
  (define-macro add1 (x)
    `(+ ,x 1)))
```

would become

```
(when-compiling
  'add1)
```

Which is equivalent to

```
add1
```

And when written at the top-level, `add1` would get eliminated in `lower-do` since it's not a `standalone?` expression.